File: go/src/prime_factors/prime_factors.go
package prime_factors

func Factors(num int) []int {
  if num == 1 {
    return []int{}
  } else {
    return FactorsExt(num, 2)
  }
}

func FactorsExt(num int, c int) []int {
  if (num <= c) {
    return []int{num}
  }

  if (num % c == 0) {
    return append([]int{c}, FactorsExt(num / c, c)...)
  } else {
    return FactorsExt(num, c + 1)
  }
}

File: scala/src/main/scala/PrimeFactors.scala
package com.crebma.primeFactors

object PrimeFactors {

  def factors(num: Int, c: Int=2) : List[Int] = {
    if (num == 1) {
      List[Int]()
    }
    else {
      if (num > c)
        if (num % c == 0) {
          List[Int](c) ++ factors(num/c, c)
        }
        else {
          factors(num, c+1)
        }
      else
      {
        List[Int](num)
      }
    }
  }
}
